amd_openvx_extensions/amd_nn/src/kernels.h

```diff
@@ -48,7 +48,12 @@ THE SOFTWARE.
 #if __APPLE__
 #include <opencl.h>
 #else
+#if ENABLE_OPENCL
 #include <CL/cl.h>
+#elif ENABLE_HIP
+#include  "../../../amd_openvx/openvx/ago/ago_internal.h"
```

Comment:
It is not a good idea to include "ago_internal.h" here. Why is this required?

Reply:
we need to have access to AgoData (i.e, it is used to cast the [parameters](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/master/amd_openvx_extensions/amd_nn/src/gather_layer.cpp#L218) for custom HIP kernel launch) which is defined in ago_internal.h.

Reply:
We need to find another way. AgoData can only be used with amdovx project. All other projects need to interface through vxapi.h or vx_amd_ext.h.

Reply:
removed including "ago_internal.h"